### DISCUSSION TITLE
Amazons own mountpoint-s3 file store with ckan

### DISCUSSION THREAD

**Discussion by brooks-eco** [ckan/ckan#8541] [Q&A]
### mountpoint-s3  https://github.com/awslabs/mountpoint-s3

AWS now have their own "high-throughput file client for mounting an Amazon S3 bucket as a local file system" ([blog article](https://aws.amazon.com/blogs/storage/the-inside-story-on-mountpoint-for-amazon-s3-a-high-performance-open-source-file-client/)).

Seems it would simplify storing ckan resources and user files on Amazon S3 compared to ckan extensions e.g. [ckanext-cloudconnector](https://github.com/DataShades/ckanext-cloudconnector) or [ckanext-s3filestore](https://github.com/keitaroinc/ckanext-s3filestore) and qld-gov-au variant  [ckanext-s3filestore](https://github.com/qld-gov-au/ckanext-s3filestore) 

Has anyone used mountpoint-s3?  comments?

We're using the [qld-gov-au ckanext-s3filestore](https://github.com/qld-gov-au/ckanext-s3filestore)  on ckan 2.11 and it is working well for us.   I suspect that mountpoint-s3 offers no operational advantage - possibly a performance advantage, but I do like the idea of removing our dependency on extensions that can be easily broken by ckan architecture/code changes (no offence @ThrawnCA). 

May also be relevant to  [files as first class citizens of ckan](https://github.com/ckan/ckan/discussions/8081) perhaps?  @wardi?

Any disadvantages?

**Comment by ThrawnCA** [comment]
:D Certainly no offence taken. Treating S3 like a regular filesystem would be useful.

I will point out one advantage of using a plugin, namely, large files can be served to the end user directly from S3, without having to pass through CKAN first.
